sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -520,18 +524,23 @@ void cleanup(PipelineOptions options) throws Exception {
 
               Job extractJob = getBigQueryServices().getJobService(bqOptions).getJob(jobRef);
 
-              Collection<String> extractFiles = null;
+              Collection<ResourceId> extractFiles = null;
               if (extractJob != null) {
                 extractFiles = getExtractFilePaths(extractDestinationDir, extractJob);
               } else {
-                IOChannelFactory factory = IOChannelUtils.getFactory(extractDestinationDir);
-                Collection<String> dirMatch = factory.match(extractDestinationDir);
-                if (!dirMatch.isEmpty()) {
-                  extractFiles = factory.match(factory.resolve(extractDestinationDir, "*"));
+                try {
+                  ResourceId extractDestinationDirResource =
+                      FileSystems.matchSingleFileSpec(extractDestinationDir).resourceId();
+                  extractFiles = ImmutableList.of(
+                      extractDestinationDirResource.resolve("*",
+                          ResolveOptions.StandardResolveOptions.RESOLVE_FILE));
+                } catch (FileNotFoundException e) {
```

Comment:
This logic seems completely broken. We match a directory (which will return at most one result, likely 0) then we add a `'*'` but don't match, instead just create a new resource name?

I think this entire code path is bogus. Why do we think there are files in the extractDestinationDir if the extract did not run?